ucsdscheduleme/Repo/ScheduleRepo.cs

```diff
@@ -31,98 +27,105 @@ public class ScheduleRepo
                     {
-                        Course = course,
-                        Section = section
+                        Course = coursesToSchedule[i],
+                        Section = j,
+                        Level = i
                     };
-                    AllSections.Add(thisSection);
-                    thisSection.IsFirstClass = thisSection.Course == firstCourse;
+
+                    thisCourse.Add(thisSection);
                 }
+                AllSections.Add(thisCourse);
             }
 
-            // Create all the edges between meetings
-            for (int i = 0; i < AllSections.Count(); i++)
+            // Create all the edges between meetings.
+            for (int i = 0; i < coursesToSchedule.Count() - 1; i++)
             {
-                for (int j = i + 1; j < AllSections.Count(); j++)
+                foreach (Node j in AllSections[i])
                 {
-                    Node node1 = AllSections[i];
-                    Node node2 = AllSections[j];
-                    if (node1.Course != node2.Course)
+                    foreach (Node k in AllSections[i + 1])
                     {
-                        if (!Conflict(node1.Section, node2.Section))
-                        {
-                            node1.Edges.Add(node2);
-                        }
+                        j.Edges.Add(k);
                     }
                 }
             }
 
-            // dfs on graph to find possible schedules
-            foreach (Node section in AllSections)
+            // Call DFS on each subgraph, starting with the first class sections
+            // being the root
+            foreach (Node i in AllSections[0])
             {
-                if (section.IsFirstClass)
-                {
-                    DFS(section, numClasses, numNodes, ref possibleSchedules);
-                }
+                DFS(i, numClasses, ref possibleSchedules);
             }
-
             return possibleSchedules;
         }
 
         /// <summary>
-        /// Performs DFS on schedule graph.
+        /// Performs DFS on schedule graph. Begins with pushing the root onto the stack. Then
+        /// continuously pops and adds the popped section onto the schedule. When a node
+        /// is popped, it will push all its children onto the stack. Once a node from the last
+        /// layer is popped, then this is a unique schedule. This schedule is checked
+        /// for conflicts and then added if there are no time conflicts.
         /// </summary>
         /// <param name="root">Node to do DFS from</param>
         /// <param name="numClasses">Total number of classes to schedule</param>
         /// <param name="possibleSchedules">Modified list of possible schedules</param>
-        private void DFS(Node root, int numClasses, int numNodes, ref List<List<Section>> possibleSchedules)
+        private void DFS(Node section, int numClasses, ref List<List<Section>> possibleSchedules)
```

Comment:
Why don't we have a null node that points to all sections in the first class, so we don't need a for loop to contain DFS

Reply:
we could do that!!

Reply:
i just figured it didn’t really make that big of a difference